core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/neo/TupleInput.scala

```diff
@@ -0,0 +1,45 @@
+package edu.uci.ics.amber.engine.architecture.worker.neo
+
+import edu.uci.ics.amber.engine.common.InputExhausted
+import edu.uci.ics.amber.engine.common.tuple.ITuple
+
+class TupleInput(batchInput: BatchInput){
```

Comment:
I feel like these functionalities can be merged into BatchInput (WorkerInternalQueue). These functions are expected from that queue.